packages/core/core/src/BundleGraph.js

```diff
@@ -41,6 +41,8 @@ type BundleGraphEdgeTypes =
   | 'internal_async';
 
 export default class BundleGraph {
+  _assetPublicIds: Set<string>;
+  _publicIdByAssetId: Map<string, string>;
```

Comment:
These need to be store on the BundleGraph because we'll need them when we merge the runtimes graph into the main bundle graph. We want to re-use any generated public id for an asset (thus the asset id => public id map), but otherwise want to quickly detect if there's a collision (the set of public ids).